### DISCUSSION TITLE
Create RelationFilterMetadata

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#6994] [Ideas]
We should be able to set default filter on relations.
One use-case is for the opportunities field, we shouldn't be able to assign a person that is from a different company than the company attached to the opportunity.
Another use-case would be to create a field FutureTasks which only show tasks in the future.

I think we should create a dedicated table `relationFiltersMetadata`.
With columns: relationMetadataId, toFieldMetadataId, operand, filterValueFieldMetadataId, filterValueRaw

fieldMetadataId is for the left side of the operand
the right side can be either a fixed value (value column) or another field in the same table (valueAsAFieldMetadataId)

In the case of an opportunities' point of contact, we would add a single Filter record with values:
- relationMetadataId: id of the opportunities <-> people relation
- toFieldMetadataId: point of contact field
- operand: eq (equal)
- filterValueFieldMetadataId: id of the company field on opportunities

This will require an additional decorator on the backend similar to `RelationMetadata` decorator.
We then need to make sure we then apply those filters automatically (doing it on the backend would be best)

(Closing https://github.com/twentyhq/twenty/issues/2632 which was related to start fresh)

**Comment by okikio** [comment]
Any progress on this?

**Reply by okikio** [reply]
I created a new GitHub Issue ticket for this https://github.com/twentyhq/twenty/issues/25579
